### PR TITLE
Add Plugin hub page listing all external plugins

### PR DESCRIPTION
**Pull request by TheStonedTurtle**
external plugins are populated by querying the current manifest.json file

This PR may require cors headers since the manifest is hosted on the rrn sub domain (repo.runelite.net)

### CODE REVIEW

File: src/routes/plugin-hub.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+          {externalPlugins.map(plugin => (
+            <ExternalPlugin key={plugin.displayName} {...plugin} />
+          ))}
+        </div>
+      </div>
+    </section>
+  </Layout>
+)
+
+const getExternalPlugins = state => state.externalPlugins
```
Comment (deathbeam) [inline src/routes/plugin-hub.js]: This should be in module, see how selectors are used everywhere else

File: src/modules/plugin-hub.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+      const pluginManifest = JSON.parse(jsonStr)
+      const plugins = pluginManifest.map(p => {
+        if (p.hasIcon) {
+          p.imageUrl = `${pluginHubUrl}${version}/${p.internalName}/${p.commit}.png`
+        } else {
+          p.imageUrl = '/img/plugin-hub/missingicon.png'
+        }
+        // Remove any tags from the description that are used for formatting inside the client
+        p.description = p.description
```
Comment (deathbeam) [inline src/modules/plugin-hub.js]: This should not be here

File: src/modules/plugin-hub.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+      // Removes the signature, and it's 4byte header, then converts the result into a string
+      const jsonStr = String.fromCharCode.apply(
+        null,
+        new Uint8Array(buffer.slice(signatureSize + 4))
+      )
+
+      const pluginManifest = JSON.parse(jsonStr)
+      const plugins = pluginManifest.map(p => {
+        if (p.hasIcon) {
+          p.imageUrl = `${pluginHubUrl}${version}/${p.internalName}/${p.commit}.png`
```
Comment (deathbeam) [inline src/modules/plugin-hub.js]: This should not be here as well. This should be handled in display component

File: src/modules/plugin-hub.js [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+        } else {
+          p.imageUrl = '/img/plugin-hub/missingicon.png'
```
Comment (deathbeam) [inline src/modules/plugin-hub.js]: see previous comment

File: src/modules/plugin-hub.js [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+import { getLatestRelease } from './bootstrap'
+
+const pluginHubUrl = 'https://repo.runelite.net/plugins/'
+
+// Actions
+export const { fetchExternalPlugins, setExternalPlugins } = createActions(
+  {
+    FETCH_EXTERNAL_PLUGINS: () => async (dispatch, getState) => {
+      const version = getLatestRelease(getState())
+      const response = await window.fetch(
```
Comment (deathbeam) [inline src/modules/plugin-hub.js]: There is api utility class for a reason, do not do this

Comment (TheStonedTurtle) [reply src/modules/plugin-hub.js]: The API utility returns the response as text or JSON and i need the response's arrayBuffer

Comment (deathbeam) [reply src/modules/plugin-hub.js]: Yeah, and file is editable :ok_hand:

File: src/modules/plugin-hub.js (hunk 3 above)
Comment (TheStonedTurtle) [reply src/modules/plugin-hub.js]: How should the display component grab the version for the url?

Comment (deathbeam) [reply src/modules/plugin-hub.js]: Yeah i guess that would be a bit annoying

**Comment by deathbeam** [comment]
Cleaned this up

**Pull request merged**